### PR TITLE
#56 convert project to astro website

### LINKED ISSUE #56: Convert to Astro website
### Problem

When CSS is modified the changes sometimes do not appear on the live website because of caching issues.

### Solution

By using Astro instead of nginx SSI includes for the headers/footers, we can build the page using Astro's static site stuff and Astro will also handle creating a different CSS file each time it changes, guaranteeing that the changes immediately appear on the deployed site.

### PR DESCRIPTION
**Pull request by SebinSong**
closes #56 

project folder structure is now similar to below screenshot according to recommendation of `astrojs` [official doc](https://docs.astro.build/en/core-concepts/project-structure/#directories-and-files).

### CODE REVIEW

**Comment by SebinSong** [comment]
@taoeffect 
Website build command `npm run build` will generate the outcome to `/dist` folder from this PR onwards. so steps for deploying the website to PROD needs to be updated accordingly too, FYI.

File: src/components/Ribbon.astro [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+---
+
+{
+  enable &&
+  <div class="ribbon-wrapper">
+    <div class="ribbon">
+      <a href="https://groupincome.org/positions/">Hiring!</a>
+    </div>
+  </div>
+}
```
Comment (taoeffect) [inline src/components/Ribbon.astro:13]: Beautifully done! Thanks for thinking of this!

**Review by taoeffect — APPROVED** [review]
@SebinSong I am really impressed with this PR. It was a big undertaking and you seem to have nailed it on the first try! 🤯 

😎 👏 

👍 

🥇

**Comment by SebinSong** [comment]
thanks @taoeffect !

**Pull request merged**